x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -193,25 +191,45 @@ export const TakeActionDropdown = React.memo(
           title: CHANGE_ALERT_STATUS,
           content: <EuiContextMenuPanel size="s" items={actionItems} />,
         },
-        /* Todo: Uncomment case action after getAddToCaseAction is split into action and modal
         {
           id: 2,
           title: ACTION_ADD_TO_CASE,
-          content: (
+          content: [
             <>
               {ecsData &&
-                timelinesUi.getAddToCaseAction({
+                timelinesUi.getAddToExistingCaseButton({
                   ecsRowData: ecsData,
                   useInsertTimeline: insertTimelineHook,
                   casePermissions,
-                  showIcon: false,
+                  appId: 'securitySolution',
+                  onClose: onCaseSelection,
                 })}
-            </>
-          ),
-        },*/
-      ],
-      [actionItems, alertsActionItems, hostIsolationAction, investigateInTimelineAction]
-    );
+            </>,
+            <>
+              {ecsData &&
+                timelinesUi.getAddToNewCaseButton({
+                  ecsRowData: ecsData,
+                  useInsertTimeline: insertTimelineHook,
+                  casePermissions,
+                  appId: 'securitySolution',
+                  onClose: onCaseSelection,
```

Comment:
nit: maybe naming: `onCaseSelection: closePopoverHandler` or `afterCaseSelection`

Reply:
ya don't mind renaming, i had just kept the click handlers and the like named what they were before